lib/inspec/resources/service.rb

```diff
@@ -537,6 +537,22 @@ def initialize(service_name, service_ctl = nil)
     end
 
     def info(service_name)
+      # `service -e` lists all enabled services. Output format:
+      # % service -e
+      # /etc/rc.d/hostid
+      # /etc/rc.d/hostid_save
+      # /etc/rc.d/cleanvar
+      # /etc/rc.d/ip6addrctl
+      # /etc/rc.d/devd
+
+      cmd = inspec.command("#{service_ctl} -e")
+      return nil if cmd.exit_status != 0
+
+      # search for the service
+
+      srv = %r{^.*/(#{service_name}$)}.match(cmd.stdout)
+      return nil if srv.nil? || srv[0].nil?
+
       # check if service is enabled
       cmd = inspec.command("#{service_ctl} #{service_name} enabled")
 
```

Comment:
This is where we'd bail to match the behaviour of `BSDInit` - instead of `enabled = cmd.exit_status == 0` this would be changed to:

```
return nil if cmd.exit_status != 0
```

Reply:
As mentioned in my [last comment](https://github.com/inspec/inspec/pull/5948#issuecomment-1078472882), we wouldn't want to bail out here, since the service should still be checked to see if it's running.